mozilla/network-pulse
# Join Us Flow

Related issue: #1092 

Form doesn't look same as what's in the mockups. For consistency I tried sticking with the same styling & form builder module that we used elsewhere on the site.

### Ideal Flow:

- visit a Pulse url with `sign_up=true` query param. e.g., https://network-pulse-staging-pr-1097.herokuapp.com/?sign_up=true
- after signing in (or creating an account) you should be auto redirected back to the Pulse page you were previously on (url should include query param `edit_new_profile=true` , e.g., https://network-pulse-staging-pr-1097.herokuapp.com/?edit_new_profile=true). Profile edit modal should pop up.

### Test Flow

Due to technical limitation with auth and a bug with local Pulse API not returning media url in absolute URL, this PR will need to be tested as the following

1. Check out this PR locally
2. Have local Pulse API and [social auth set up](https://github.com/mozilla/network-pulse-api#setting-up-social-authentication)
3. Use the following env vars in Pulse front-ends `.env`
```
PULSE_API=http://test.example.com:8000/api/pulse
PULSE_LOGIN_URL=http://test.example.com:8000/accounts/login/
PULSE_LOGOUT_URL=http://test.example.com:8000/accounts/logout/
```
4. Run Pulse API and Pulse locally
5. Go to http://test.example.com:3000/latest?sign_up=true
6. If you are not signed in, sign in with Google
<img width="250" alt="image" src="https://user-images.githubusercontent.com/2896608/59374014-bbfd4e80-8cff-11e9-9a05-a4b42310c9f4.png">

7. You should now be redirected to http://test.example.com:3000/latest?edit_new_profile=true and modal should pop up.
8. Go through the profile edit flow in modal and see if there's any issue. (`Continue` button triggers form validation whereas `Skip` button does not)

Note: there's a bug on API's end about not sending media URL in absolute format, and hence the default thumbnail can't be displayed properly. See screencap below

<img width="500" alt="image" src="https://user-images.githubusercontent.com/2896608/59388315-4e154f00-8d20-11e9-9428-8b5ec583835d.png">

### Check out how the modal and form work using review app

1. make sure you are logged into staging Pulse API
  - you can do it by going to the review app https://network-pulse-staging-pr-1097.herokuapp.com
  - use the sign in/up link at top right
2. visit any page on this review app with query param `edit_new_profile=true` (e.g., https://network-pulse-staging-pr-1097.herokuapp.com/featured?edit_new_profile=true)
3. play around with the modal and form (form won't submit due to staging API blocking review apps from POSTing)
